From 1a8b967a8d4c35457719d94137f2695967a1f7d8 Mon Sep 17 00:00:00 2001
From: JPyke3 <github@pyk.ee>
Date: Sun, 2 Aug 2026 22:24:43 +1000
Subject: [PATCH] steamcompmgr: work around ARM64 Steam night mode property

---
 src/steamcompmgr.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/steamcompmgr.cpp b/src/steamcompmgr.cpp
index b74b8a7e..f7fb9895 100644
--- a/src/steamcompmgr.cpp
+++ b/src/steamcompmgr.cpp
@@ -7394,9 +7394,17 @@ handle_property_notify(xwayland_ctx_t *ctx, XPropertyEvent *ev)
 		}
 
 		nightmode_t nightmode;
+		
+		#if defined (__aarch64__)
+		nightmode.amount = vec[0];
+		nightmode.hue = 25.0f / 360.0f;
+		nightmode.saturation = vec[1];
+		#else
 		nightmode.amount = vec[0];
 		nightmode.hue = vec[1];
 		nightmode.saturation = vec[2];
+		#endif
+
 
 		if ( set_color_nightmode( nightmode ) )
 			hasRepaint = true;
